### PR TITLE
Initial functionality

### PR DESCRIPTION
**Pull request by dav3r**
#  #

## 🗣 Description ##

This PR implements the initial functionality of this library, which is to fetch CVE data from a source and sync it with a CyHy database.

## 💭 Motivation and context ##

The functionality in this PR is very similar to the [legacy cyhy-nvdsync](https://github.com/cisagov/cyhy-core/blob/develop/bin/cyhy-nvdsync), however it uses the new Python 3 [cyhy-db](https://github.com/cisagov/cyhy-db) and [cyhy-config](https://github.com/cisagov/cyhy-config) libraries.

## 🧪 Testing ##

This code has been 100% covered by unit tests.  I also confirmed that the entire set of [CVE JSON files from the NVD](https://nvd.nist.gov/vuln/data-feeds#JSON_FEED) can be successfully ingested into a CyHy database using this code.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [x] Tests have been added and/or modified to cover the changes in this PR.
- [x] All new and existing tests pass.

## ✅ Pre-merge checklist ##

- [x] Finalize version.

## ✅ Post-merge checklist ##

- [x] Create `v1.0.0` release.
- [x] Create `v1` tag (`git tag v1 v1.0.0; git push --tags`)

### CODE REVIEW

**Review by dv4harr10 — APPROVED** [review]

**Review by jsf9k — APPROVED** [review]
I also verified that I could run the `pytest` tests locally without any issues.

File: src/cyhy_cvesync/cve_sync.py [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+MAX_CVE_URL_RETRIES = 10
+
+# Map to track existing CVE documents that were not updated
+cve_map: Dict[str, CVEDoc] = {}
+
+logger = logging.getLogger(f"{CYHY_ROOT_LOGGER}.{__name__}")
+
+
+async def process_cve_json(cve_json: dict) -> Tuple[int, int]:
+    """Process the provided CVEs JSON and update the database with their contents."""
```
Comment (felddy) [inline src/cyhy_cvesync/cve_sync.py]: Document the contents of the return value tuple.

File: src/cyhy_cvesync/cve_sync.py [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+        with gzip.GzipFile(fileobj=BytesIO(response_content)) as f:
+            return json.loads(f.read().decode("utf-8"))
+    else:
+        return json.loads(response_content)
+
+
+async def process_urls(
+    cve_urls: List[str], cve_data_gzipped: bool
+) -> Tuple[int, int, int]:
+    """Process URLs containing CVE data."""
```
Comment (felddy) [inline src/cyhy_cvesync/cve_sync.py]: Return value documentation please.

**Review by felddy — APPROVED** [review]
Looks good.  Just need a couple comments for return values.

File: src/cyhy_cvesync/cve_sync.py (hunk 1 above)
Comment (dav3r) [reply src/cyhy_cvesync/cve_sync.py]: Please see https://github.com/cisagov/cyhy-cvesync/pull/1/commits/ce8375a815270fe3a06316ec8ab2f1637d72bca1 and resolve this conversation if that works for you.

File: src/cyhy_cvesync/cve_sync.py (hunk 2 above)
Comment (dav3r) [reply src/cyhy_cvesync/cve_sync.py]: Please see https://github.com/cisagov/cyhy-cvesync/pull/1/commits/ce8375a815270fe3a06316ec8ab2f1637d72bca1 and resolve this conversation if that works for you.

**Pull request merged**